pkg/kubelet/kubelet.go

```diff
@@ -1779,23 +1782,31 @@ func (kl *Kubelet) canAdmitPod(pods []*api.Pod, pod *api.Pod) (bool, string, str
 // state every sync-frequency seconds. Never returns.
 func (kl *Kubelet) syncLoop(updates <-chan PodUpdate, handler SyncHandler) {
 	glog.Info("Starting kubelet main sync loop.")
+	var housekeepingTimestamp time.Time
 	for {
+		if !kl.containerRuntimeUp() {
+			time.Sleep(5 * time.Second)
+			glog.Infof("Skipping pod synchronization, container runtime is not up.")
+			continue
+		}
+		if !kl.doneNetworkConfigure() {
+			time.Sleep(5 * time.Second)
+			glog.Infof("Skipping pod synchronization, network is not configured")
+			continue
+		}
+		if housekeepingTimestamp.IsZero() || time.Since(housekeepingTimestamp) > housekeepingPeriod {
```

Comment:
I don't want the housekeeping to happen too often (on every event/iteration), so I use the timestamp as an indication. How would util.Until help in this case?